Framework/Geometry/inc/MantidGeometry/Instrument/ComponentInfo.h

```diff
@@ -54,19 +54,24 @@ class MANTID_GEOMETRY_DLL ComponentInfo {
   /// Pointer to the actual ComponentInfo object (non-wrapping part).
   std::unique_ptr<Beamline::ComponentInfo> m_componentInfo;
   /// Collection of component ids
-  boost::shared_ptr<const std::vector<Mantid::Geometry::IComponent *>>
-      m_componentIds;
+  boost::shared_ptr<const std::vector<Geometry::IComponent *>> m_componentIds;
   /// Map of component ids to indexes
   boost::shared_ptr<const std::unordered_map<Geometry::IComponent *, size_t>>
       m_compIDToIndex;
 
+  /// Shapes for each component
+  boost::shared_ptr<std::vector<boost::shared_ptr<const Geometry::Object>>>
+      m_shapes;
```

Comment:
This looks like it will be pointing to the shapes in the old instrument tree? Maybe it would make sense to store all unique shapes in `ComponentInfo` and have raw pointers or indices into a vector of shapes here to avoid the `shared_ptr` overhead?

But I suppose having such a vector of unique shapes is not so trivial/cheap to set up?